### PR TITLE
Making Spring AI Optional

### PR DESCRIPTION
**Pull request by raphaeldelio**
Spring AI is not in a release version yet. Given that there are companies that don't allow the use of milestone versions in their projects, they lost the ability of upgrading to the latest version of Redis OM Spring. To give them back the ability to keep using the latest version of Redis OM Spring, it was decided that the dependencies related to Spring AI were going to become optional.

To make them optional, we have two options: 
1. Set the specific dependencies to "optional" in the Maven file. This will require that anyone who wants to use AI features will have to explicitly declare these dependencies. 

Example in Gradle KTS:

```
	maven {
		name = "Spring Milestones"
		url = uri("https://repo.spring.io/milestone")
		mavenContent {
			snapshotsOnly()
		}
	}
```

```
	implementation(platform("org.springframework.ai:spring-ai-bom:1.0.0-M2"))

	implementation("com.redis.om:redis-om-spring:0.9.7")
	implementation("org.springframework.ai:spring-ai-openai")
	implementation("jakarta.websocket:jakarta.websocket-api:2.1.1")
	implementation("jakarta.websocket:jakarta.websocket-client-api:2.1.1")
	implementation("org.springframework.ai:spring-ai-ollama")
	implementation("org.springframework.ai:spring-ai-azure-openai")
	implementation("org.springframework.ai:spring-ai-vertex-ai-palm2")
	implementation("org.springframework.ai:spring-ai-bedrock")
	implementation("org.springframework.ai:spring-ai-transformers")
	implementation("org.springframework.ai:spring-ai-mistral-ai")
	implementation("org.springframework.ai:spring-ai-minimax")
	implementation("org.springframework.ai:spring-ai-zhipuai")
```

3. Leave the dependencies as they are and allow those who don't want them to explicitly exclude them. 

Example in Gradle KTS:
```
	implementation("com.redis.om:redis-om-spring:0.9.7"){
		exclude(group = "org.springframework.ai")
		exclude(group = "ai.djl")
		exclude(group = "ai.djlpytorch")
		exclude(group = "ai.djl.spring")
		exclude(group = "ai.djl.huggingface")
	}
```

Regardless of the choice we decide to go with, we will need to make sure that the Beans related to AI features are only created in case these dependencies are added to the project. To do so, I decided to:

- Split the RedisModulesConfiguration into itself and RedisAiConfiguration. 
This allowed me to set a Conditional to the whole configuration class. Unfortunately, adding the condition to the beans itself wasn't an option because SpringBoot would load the whole Configuration class before evaluating the condition for each Bean and resulting in a ClassNotFoundException. 

Because of this same reason, I decided to split the RedisOMProperties into itself and RedisOMAiProperties too. 

We cannot have two Configuration classes with the same prefix. This forced me to use "redis.om.spring.ai" as the prefix for the new RedisOMAiProperties class. 

Besides that I also added a warning to BasicSentinelTest because I spent some time trying to figure out why it was failing until I finally understood that for this specific test I had to manually start the Redis Container.

Moreover, I didn't know if the name or location of the new files are correct. 

I tested this by loading Redis OM Spring library in a separate project and testing it with and without the dependencies. I couldn't find a way to test it without the dependencies with the existing testing configuration. 

When running the project without the required dependencies and the conditional set to true it will throw a ClassNotFoundException at startup due to the missing classes during Bean creation. 

If running without the required dependencies and the conditional set to false, it will normally start.

### CODE REVIEW

**Comment by raphaeldelio** [comment]
All tests are now passing locally. Turns out that the optional tag wasn't the issue, the Embedder bean had to be separate into the existing and the new Configuration classes as well. 

The existing class will create it as a NoopEmbedder while the new one will create it as a DefaultEmbedder.

File: demos/roms-vss/src/main/resources/application.properties
Comment (bsbodden) [inline demos/roms-vss/src/main/resources/application.properties:1]: Let's remove the `.djl.` from the property name

**Review by bsbodden — CHANGES_REQUESTED** [review]
Just a small change... removing the .djl. from the property

File: demos/roms-vss/src/main/resources/application.properties
Comment (raphaeldelio) [reply demos/roms-vss/src/main/resources/application.properties:1]: done!

**Review by bsbodden — APPROVED** [review]
LGTM!

**Pull request merged**